src/components/atoms/icon-box/IconBox.tsx

```diff
@@ -15,6 +15,13 @@ export type IconBoxPropsType = {
      * The color of this icon box. When not provided, it will default to #787878 (medium gray)
      */
     color: string,
+
+    /**
+     * If provided the className used for the icon box are overwritten.
+     */
+    // The need for this might indicate that this component isn't atomic 
+    // enough / is too specific and might need refractor or a new component to be made.
+    classNameOverride?: string 
```

Comment:
Based on usage could this not be reduced down into a type ie `type: 'normal' | 'collapsed' | 'expanded'` and then base the class on that which prevents us just overwriting the class name for the concerns you've mentioned

Reply:
The problem is that the IconBox CSS is too specific to a particular usage (it has lots of fixed values etc.) and I needed a way to overwrite that.

I think having normal/collapsed/expanded variants is a great idea but they need to be made in a way that is more flexible in usage.

Reply:
It was designed to be a general component because it's just an icon in a padded box. I can see the need for a collapsed version because the padding is quite large normally (was designed for indicators ie the example below)

![image](https://user-images.githubusercontent.com/9435503/87248366-03996100-c451-11ea-8e2e-7d1d7984af07.png)

I would highly recommend making this one take a type and collapsing and expanding it that way rather than overwriting the class because to me that defeats the object of the component itself. Alternatively we just reimplement the icon box in the venue page if it requires that many modifications because at that point it is a different style in a lot of ways. Alignment and stuff has so far generally be done with flex which means that fixed margins on it worked fine.

One other final adjustment that could be done is to take a style prop and pass it in to the container. That would allow overrides to be passed in from the parent in the rendering code rather than having to define new styles and stuff.

Reply:
I had to overwrite the class as the existing CSS would not allow it to be used as needed (as in adding more over the top was insufficient), we could reimplement a new iconbox but I think its a fairly general component so it should be refactored in a way it can be used for both.